src/plugins/inspector/public/views/requests/components/details/req_details_request_content.tsx

```diff
@@ -0,0 +1,123 @@
+const openInConsoleLabel = i18n.translate('inspector.requests.openInConsoleLabel', {
+  defaultMessage: 'Open in Console',
+});
+
+const openInSearchProfilerLabel = i18n.translate('inspector.requests.openInSearchProfilerLabel', {
+  defaultMessage: 'Open in Search Profiler',
+});
+
+interface RequestDetailsRequestContentProps {
+  indexPattern?: string;
+  requestParams?: ConnectionRequestParams;
+  json: string;
+}
+
+export const RequestDetailsRequestContent: React.FC<RequestDetailsRequestContentProps> = ({
+  requestParams,
+  indexPattern,
+  json,
+}) => {
+  const { services } = useKibana<InspectorKibanaServices>();
+
+  function getValue(): string {
+    if (!requestParams) {
+      return json;
+    }
+
+    const fullPath = requestParams.querystring
+      ? `${requestParams.path}?${requestParams.querystring}`
+      : requestParams.path;
+
+    return `${requestParams.method} ${fullPath}\n${json}`;
+  }
+
+  const value = getValue();
+
+  const navigateToUrl = services.application?.navigateToUrl;
+
+  // "Open in Console" button
+  const devToolsDataUri = compressToEncodedURIComponent(value);
+  const consoleHref = services.share.url.locators
+    .get('CONSOLE_APP_LOCATOR')
+    ?.useUrl({ loadFrom: `data:text/plain,${devToolsDataUri}` });
+  // Check if both the Dev Tools UI and the Console UI are enabled.
+  const canShowDevTools =
+    services.application?.capabilities?.dev_tools.show && consoleHref !== undefined;
+  const shouldShowDevToolsLink = !!(requestParams && canShowDevTools);
+  const handleDevToolsLinkClick = useCallback(
+    () => consoleHref && navigateToUrl && navigateToUrl(consoleHref),
+    [consoleHref, navigateToUrl]
+  );
+
+  // "Open in Search Profiler" button
+  const searchProfilerDataUri = compressToEncodedURIComponent(json);
+  const searchProfilerHref = services.share.url.locators
+    .get('SEARCH_PROFILER_LOCATOR')
+    ?.useUrl({ index: indexPattern, loadFrom: `data:text/plain,${searchProfilerDataUri}` });
+  // Check if both the Dev Tools UI and the SearchProfiler UI are enabled.
+  const canShowsearchProfiler =
+    services.application?.capabilities?.dev_tools.show && searchProfilerHref !== undefined;
+  const shouldShowsearchProfilerLink = !!(indexPattern && canShowsearchProfiler);
+  const handleSearchProfilerLinkClick = useCallback(
+    () => searchProfilerHref && navigateToUrl && navigateToUrl(searchProfilerHref),
+    [searchProfilerHref, navigateToUrl]
+  );
+
+  const actions: Array<{ name: string; action: ReactNode }> = [];
+
+  if (shouldShowDevToolsLink) {
+    actions.push({
+      name: 'openInConsole',
+      action: (
+        <EuiButtonEmpty
+          size="xs"
+          flush="right"
+          iconType="wrench"
+          href={consoleHref}
+          onClick={handleDevToolsLinkClick}
+          data-test-subj="inspectorRequestOpenInConsoleButton"
+        >
+          {openInConsoleLabel}
+        </EuiButtonEmpty>
+      ),
+    });
+  }
+
+  if (shouldShowsearchProfilerLink) {
```

Comment:
nit: `shouldShowsearchProfilerLink` camel case should be `shouldShowSearchProfilerLink`